### DISCUSSION TITLE
QuestPDF - 支援中文、免費好用的 C# PDF 產生器 | 全端開發人員天梯

### DISCUSSION THREAD

**Discussion by giscus** [wellwind/wellwind.github.io#82] [Announcements]
# QuestPDF - 支援中文、免費好用的 C# PDF 產生器 | 全端開發人員天梯

產生 PDF 的套件並不難找，有趣的是，.NET 的 PDF 套件雖然不少，但幾乎都是商業授權，免費又好用的卻不多，最近找到一套 QuestPDF，它的 API 設計非常漂亮，支援中文，並且免費開源，雖然不一定是最強大的套件，但對於一般產生 PDF 的工作，已經非常好用了。

https://fullstackladder.dev/blog/2023/02/05/introduction-quest-pdf/

**Comment by sense10240** [comment]
很好的介紹，可惜QuestPDF 並不是完全免費，個人或公司營收低於1M USD才能免費使用

**Reply by wellwind** [reply]
看到這則留言真是嚇壞我了 😱

當時在撰寫這篇文章時，我沒看到任何需要付費的內容，完全就是 MIT 授權

剛剛再去看了一下，竟然真的出現了 Pricing 頁面！

但從下面的 「Frequently Asked Questions」來看是 2023.X 版本之後才有商用問題，使用之前的版本依然是免費，只要不破壞 MIT 授權規則就可以了！

> What library version does the commercial license apply to?
> The commercial license applies to the QuestPDF 2023.X releases and beyond. All versions before are available completly for free under the MIT license.

關於文章內容，我再更新補充說明一下，感謝提醒 🙏

**Reply by sense10240** [reply]
太好了，我真的沒留意到FAQ部份，我在煩支援跨平台PDF生成套件，我馬上下載 2022.12.6 版本試試看，感謝你!!!

**Comment by blueberry633** [comment]
感謝分享，非常好用。
之前使用的是 PDFsharp，多數的時候須要像輸出點陣圖這樣，一個一個位置來定位元素。
https://www.pdfsharp.net/
